File: app/src/main/java/com/example/alexandra/bigriverbookstore/resources/Book.java
package com.example.alexandra.bigriverbookstore.resources;

import com.squareup.moshi.Json;
import moe.banana.jsonapi2.HasMany;
import moe.banana.jsonapi2.HasOne;
import moe.banana.jsonapi2.JsonApi;
import moe.banana.jsonapi2.Resource;

import java.util.List;

@JsonApi(type = "books")
public class Book extends Resource {
    public String title;
    @Json(name = "date_published")
    public String datePublished;
    public String isbn;
    private HasOne<Author> author;
    private HasMany<Photo> photos;

    public Author getAuthor() {
        return author.get(getDocument());
    }

    public List<Photo> getAllPhotos() {
        return photos.get(getDocument());
    }
}

File: app/src/main/java/com/example/alexandra/bigriverbookstore/Book.kt
package com.example.alexandra.bigriverbookstore

import com.example.alexandra.bigriverbookstore.resources.Book

data class Book(val name: String, val author: String, val picturePath: String) {
    companion object {
        fun from(book: Book): Book {
            val name = book.title
            val author =  book.author.name
            val picturePath = book.allPhotos.first().uri

            return Book(name, author, picturePath)
        }
    }
}